### PR TITLE
Create New Book Request

### PR DESCRIPTION
**Pull request by Rownaknira**

### CODE REVIEW

File: app/views/requests/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<div class="row">
+  <h1><%= t ".request.new" %></h1>
+  <br>
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for @request do |f| %>
+      <div class="form-group">
+        <%= f.label :title %> 
+        <%= f.text_field :title, class: 'form-control' %>
```
Comment (SSalekin) [inline app/views/requests/new.html.erb]: double qoute

File: app/views/requests/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      </div>
+      <div class="form-group">
+        <%= f.label :author %> 
+        <%= f.text_field :author, class: 'form-control' %>
```
Comment (SSalekin) [inline app/views/requests/new.html.erb]: double qoute

**Comment by SSalekin** [comment]
ok.

**Comment by noor-framgia** [comment]
ok

File: app/controllers/user/requests_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class User::RequestsController < ApplicationController
```
Comment (dieunb) [inline app/controllers/user/requests_controller.rb]: I think we don't need namespace user.

File: app/controllers/requests_controller.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+class RequestsController < ApplicationController
+  def new
+    @request = Request.new
```
Comment (dieunb) [inline app/controllers/requests_controller.rb]: should implement `cancancan` early to remove this line code

File: app/controllers/requests_controller.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def create
+    @request = Request.new request_params
```
Comment (dieunb) [inline app/controllers/requests_controller.rb]: should implement `cancancan` early to remove this line code

File: app/views/requests/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<div class="row">
+  <h1><%= t ".title" %></h1>
+  <br>
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for @request do |f| %>
+      <div class="form-group">
+        <%= f.label :title %>
```
Comment (dieunb) [inline app/views/requests/new.html.erb]: only `title` and `author` for request? Another field, example: book_link, description, ...

File: config/routes.rb [hunk 7]
```diff
@@ -1,4 +1,7 @@
 Rails.application.routes.draw do
+
+  resources :requests
```
Comment (dieunb) [inline config/routes.rb]: use `only` or `except` to manage action what you want to use.

File: app/controllers/requests_controller.rb (hunk 4 above)
Comment (Rownaknira) [inline app/controllers/requests_controller.rb]: ok. Then I implement cancancan first, then change this code later

File: db/schema.rb [hunk 8]
```diff
@@ -84,13 +84,14 @@
...
-    t.string   "author",     limit: 255
-    t.integer  "link",       limit: 4
-    t.integer  "status",     limit: 4
-    t.integer  "user_id",    limit: 4
-    t.datetime "created_at",             null: false
-    t.datetime "updated_at",             null: false
+    t.string   "title",       limit: 255
+    t.string   "author",      limit: 255
+    t.string   "link",        limit: 255
+    t.string   "description", limit: 255
```
Comment (dieunb) [inline db/schema.rb]: should `t.text`

File: db/migrate/20151230075901_create_requests.rb [hunk 9]
```diff
@@ -3,7 +3,8 @@ def change
     create_table :requests do |t|
       t.string :title
       t.string :author
-      t.integer :link
+      t.string :link
+      t.string :description
```
Comment (dieunb) [inline db/migrate/20151230075901_create_requests.rb]: should `t.text`

**Pull request merged**